### PR TITLE
bug 1827748: feature: opm (index|registry) prune command

### PR DESCRIPTION
**Pull request by djzager**
**Description of the change:**

This makes it possible for a user to take an index image or a database
referencing a large collection of operator bundles to filter down the
collection.

**Motivation for the change:**

Make it possible to scale down operator collections.

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: cmd/opm/index/filter.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	indexCmd.Flags().Bool("debug", false, "enable debug logging")
+	indexCmd.Flags().Bool("generate", false, "if enabled, just creates the dockerfile and saves it to local disk")
+	indexCmd.Flags().StringP("out-dockerfile", "d", "", "if generating the dockerfile, this flag is used to (optionally) specify a dockerfile name")
+	indexCmd.Flags().StringP("from-index", "f", "", "previous index to filter")
+	if err := indexCmd.MarkFlagRequired("from-index"); err != nil {
+		logrus.Panic("Failed to set required `from-index` flag for `index filter`")
+	}
+	indexCmd.Flags().StringSliceP("operators", "o", nil, "comma separated list of operators to filter")
```
[the remark this answers is not in the document]
Comment (djzager) [reply cmd/opm/index/filter.go:33]: I agree that "packages" may be the least confusing. The problem for me is that I took these flags off of `cmd/opm/index/delete.go` and so I'm hesitant to 1) mess with what already exists 2) change flags just for the filter command.

File: cmd/opm/registry/rm.go [hunk 2]
```diff
@@ -25,7 +25,10 @@ func newRegistryRmCmd() *cobra.Command {
 
 	rootCmd.Flags().Bool("debug", false, "enable debug logging")
 	rootCmd.Flags().StringP("database", "d", "bundles.db", "relative path to database file")
-	rootCmd.Flags().StringSliceP("packages", "o", []string{}, "comma separated list of package names to be deleted")
+	rootCmd.Flags().StringSliceP("packages", "o", nil, "comma separated list of package names to be deleted")
```
Comment (djzager) [inline cmd/opm/registry/rm.go:28]: I need to pull this out into a separate PR.

**Comment by djzager** [comment]
/hold

I broke something. Will fix and add appropriate testcase.

**Comment by ecordell** [comment]
/lgtm

Thanks for the wait @djzager!

This looks good modulo some refactoring that we need to do across the entire indexer.

Just so you know, we have to either file a BZ for this or wait for master to open (because we currently build upstream and downstream from this repo).
